drivers/mtd_sdcard/mtd_sdcard.c

```diff
@@ -150,13 +184,40 @@ static int mtd_sdcard_power(mtd_dev_t *dev, enum mtd_power_state power)
     return -ENOTSUP; /* currently not supported */
 }
 
+static int mtd_sdcard_read(mtd_dev_t *dev, void *buff, uint32_t addr,
+                           uint32_t size)
+{
+    int res = mtd_sdcard_read_page(dev, buff, addr / SD_HC_BLOCK_SIZE,
+                                   addr % SD_HC_BLOCK_SIZE, size);
+    if (res < 0) {
+        return res;
+    }
+    if (res == (int)size) {
+        return 0;
+    }
+    return -EOVERFLOW;
+}
+
+static int mtd_sdcard_write(mtd_dev_t *dev, const void *buff, uint32_t addr,
+                            uint32_t size)
+{
+    int res =  mtd_sdcard_write_page(dev, buff, addr / SD_HC_BLOCK_SIZE,
+                                     addr % SD_HC_BLOCK_SIZE, size);
+    if (res < 0) {
+        return res;
+    }
+    if (res == (int)size) {
+        return 0;
+    }
+    return -EOVERFLOW;
+}
+
 const mtd_desc_t mtd_sdcard_driver = {
     .init = mtd_sdcard_init,
     .read = mtd_sdcard_read,
     .read_page = mtd_sdcard_read_page,
     .write = mtd_sdcard_write,
     .write_page = mtd_sdcard_write_page,
-    .erase = mtd_sdcard_erase,
+    .erase_sector = mtd_sdcard_erase_sector,
```

Comment:
Why remove the `erase` function? Isn't it mostly the same function but with `addr` as input instead of sector number? (can't we have both)

Reply:
The MTD layer will [take care of this](https://github.com/RIOT-OS/RIOT/blob/master/drivers/mtd/mtd.c#L281) - no need to implement this twice in the driver. And `sdcard_spi` only supports sector erase to begin with.

Reply:
Thanks for clarifying!